Remove unnecessary check for Qt version.

$ diff -urN SLiM/QtSLiM/main.cpp.orig SLiM/QtSLiM/main.cpp
--- SLiM/QtSLiM/main.cpp.orig	2022-02-12 17:07:41.000000000 +0100
+++ SLiM/QtSLiM/main.cpp	2022-03-10 14:31:11.387185200 +0100
@@ -192,13 +192,6 @@
 
 int main(int argc, char *argv[])
 {
-    // Check that the run-time Qt version matches the compile-time Qt version
-    if (strcmp(qVersion(), QT_VERSION_STR) != 0)
-    {
-        std::cout << "Run-time Qt version " << qVersion() << " does not match compile-time Qt version " << QT_VERSION_STR << std::endl;
-        exit(EXIT_FAILURE);
-    }
-    
     // Check for running under ASAN and log to confirm it is enabled; see SLiM.pro to enable it
 #if defined(__has_feature)
 #  if __has_feature(address_sanitizer)
